crates/ra_syntax/src/tests.rs

```diff
@@ -10,7 +10,8 @@ use crate::{fuzz, SourceFile};
 #[test]
 fn lexer_tests() {
     dir_tests(&test_data_dir(), &["lexer"], |text, _| {
-        let tokens = crate::tokenize(text);
+        // FIXME: add tests for errors (their format is up to discussion)
+        let tokens = crate::tokenize(text).tokens;
```

Comment:
Your input is very important here, guys!

Reply:
You could probably just update [`dump_tokens`](https://github.com/rust-analyzer/rust-analyzer/blob/d1330a4a65f0113c687716a5a679239af4df9c11/crates/ra_syntax/src/tests.rs#L104-L115) to take `ParsedTokens` and then also output the errors like we do the tokens. If you don't print anything when there's no errors, you shouldn't even need to touch the existing tests.

Reply:
Just dumping errors in `dump_tokens` after the tokens themselves seems good to me.

Unrelated, but perhaps a  slightly more information rich way to write this would be:

```rust
let ParsedToken { tokens, errors: _ } = crate::tokenize(text)
```

Ie, explicitely naming the thing you are ignoring (which usually is important for erroes).

Reply:
Okay guys, will do!
Also, @matklad, good point on explicit errors dropping. Do you think we should use this pattern in all places `tokenize*()/single_token()/first_token()` are called?

Reply:
I think so! In general, I just always try to destruct tuples that way:

```
13:49:36|~/projects/rust-analyzer|HEAD✓
λ rg  'let .*, _\w+\)'
xtask/src/codegen/gen_syntax.rs
167:    let punctuation_values = grammar.punct.iter().map(|(token, _name)| {

crates/ra_hir_def/src/generics.rs
59:        let (params, _source_map) = GenericParams::new(db, def.into());

crates/ra_hir/src/code_model.rs
1072:        let (adt, _subst) = self.ty.value.as_adt()?;

crates/ra_batch/src/lib.rs
53:    let (crate_graph, _crate_names) =
147:        let (host, _roots) = load_cargo(path).unwrap();

crates/ra_hir_ty/src/utils.rs
141:            let (_total, parent_len, _child) = self.len_split();

crates/ra_parser/src/grammar/expressions.rs
27:    let (cm, _block_like) = expr(p);

crates/ra_hir_def/src/nameres/collector.rs
947:        let (db, _file_id) = TestDB::with_single_file(&code);

crates/ra_hir_ty/src/infer/coerce.rs
253:                let (last_field_id, _data) = fields.next_back()?;

crates/ra_parser/src/grammar/expressions/atom.rs
560:    let (completed, _is_block) =
``